### PR TITLE
Trying jdbc-socket-factory-core in the Libraries BOM

### PR DESCRIPTION
**Pull request by suztomo**
Checking whether jdbc-socket-factory-core breaks anything.

CC: @meltsufin @elharo 

The repository (https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory) seems to publish many artifacts. 

https://search.maven.org/search?q=g:com.google.cloud.sql

If we decide to include it, we can ask them to publish a BOM for these artifacts.

### CODE REVIEW

**Comment by meltsufin** [comment]
Yes, if we were to include it, we need the other components as well, not just the core.
cc/ @shubha-rajan.

**Review by meltsufin — APPROVED** [review]

File: boms/cloud-oss-bom/pom.xml
Comment (elharo) [inline boms/cloud-oss-bom/pom.xml:272]: Does this project have its own BOM? Should i?

**Review by elharo — CHANGES_REQUESTED** [review]
Let's discuss this further and make certain we want/need this.

File: boms/cloud-oss-bom/pom.xml
Comment (suztomo) [reply boms/cloud-oss-bom/pom.xml:272]: No, it doesn't have a BOM. I can ask them to release one.

**Comment by suztomo** [comment]
Created an issue for cloud-sql-jdbc-socket-factory https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/issues/499.

**Comment by suztomo** [comment]
The libraries seem to include `xerces:xerces-impl:2.6.2` in their dependency tree. Probably optional scope in transitive dependencies. The nonexistent artifact is causing the check failure:

```
Error:  Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 443.367 s <<< FAILURE! - in com.google.cloud.MaximumLinkageErrorsTest
Error:  testForNewLinkageErrors(com.google.cloud.MaximumLinkageErrorsTest)  Time elapsed: 443.314 s  <<< ERROR!
java.lang.NullPointerException
	at com.google.cloud.MaximumLinkageErrorsTest.testForNewLinkageErrors(MaximumLinkageErrorsTest.java:63)

```

**Comment by elharo** [comment]
If  jdbc-socket-factory-core wants to be in the BOM, they should fix this.

**Comment by suztomo** [comment]
cloud-sql-connector-r2dbc-core has ~7000 dependency nodes (including optional).

https://gist.github.com/suztomo/17db9bca5697a34b6076286b3085ed3a

 Some of the artifacts (`xerces:xerces-impl:jar:2.6.2`, `xml-apis:xml-apis:jar:2.6.2`, etc.) are not available in Maven Central. Therefore the tool in this repository cannot handle them as of now.

**Comment by shubha-rajan** [comment]
Hi @suztomo, we just released a new version of the Java Connectors that should be compatible with the libraries BOM.  Please let us know if it works or if we need to fix anything. Thanks!

File: boms/cloud-oss-bom/pom.xml
Comment (shubha-rajan) [inline boms/cloud-oss-bom/pom.xml:60]:
```suggestion
    <cloud.sql.jdbc.socket.factory.version>1.3.1</cloud.sql.jdbc.socket.factory.version>
```

Comment (suztomo) [reply boms/cloud-oss-bom/pom.xml]: Done

**Review by suztomo — COMMENTED** [review]
@shubha-rajan We recently started revising the strictness of the dependency management in this BOM, which requires the participating libraries to have the same set of dependencies. I'll send you separate communication. Until we figure out the course, we don't merge this PR.

**Comment by suztomo** [comment]
The build fails.

```
[INFO] Running com.google.cloud.MaximumLinkageErrorsTest
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Jul 14, 2021 5:23:41 PM com.google.cloud.tools.opensource.classpath.LinkageChecker create
SEVERE: org.eclipse.swt:org.eclipse.swt.gtk.linux.x86_64:jar:4.5.2 was not resolved. Dependency path: null > com.google.cloud.sql:cloud-sql-connector-r2dbc-mysql:jar:1.3.1 (compile) > dev.miku:r2dbc-mysql:jar:0.8.2.RELEASE (provided) > io.projectreactor.addons:reactor-extra:jar:3.3.3.RELEASE (compile) > org.eclipse.swt:org.eclipse.swt.gtk.linux.x86_64:jar:4.5.2 (compile?)
Error:  Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 180.305 s <<< FAILURE! - in com.google.cloud.MaximumLinkageErrorsTest
Error:  testForNewLinkageErrors(com.google.cloud.MaximumLinkageErrorsTest)  Time elapsed: 180.221 s  <<< ERROR!
java.io.IOException: Could not resolve 1 dependency. See the message above for details.
	at com.google.cloud.MaximumLinkageErrorsTest.testForNewLinkageErrors(MaximumLinkageErrorsTest.java:52)
```

**Pull request closed without merging**
